src/api/java/baritone/api/utils/Helper.java

```diff
@@ -69,4 +69,9 @@ default void logDirect(String message) {
         component.appendSibling(new TextComponentString(" " + message));
         Minecraft.getMinecraft().addScheduledTask(() -> BaritoneAPI.getSettings().logger.value.accept(component));
     }
+    default void logDirectTextComponentString(TextComponentString message) {
```

Comment:
is this really required...

Reply:
I use it to have different and multiple colors.
I use yellow in logic messages. see the demo.

and no, the default Minecraft color system adds a character.